### DISCUSSION TITLE
Proposal for the ability to populate the ViewModel in the View's BindingContext with ServiceCollectionExtensions.

### DISCUSSION THREAD

**Discussion by StdEnku** [CommunityToolkit/Maui#636] [New Feature Discussions]
Hello

Thank you for providing us with a very simple and easy-to-use library.

Currently, it is possible to register a View and a ViewModel in a DI container at the same time using the extension method defined in the ServiceCollectionExtensions, but I would like to have the ability to register a ViewModel instance in the View's BindingContext in that method.

Specifically, this could be accomplished by modifying all the extension methods as shown in the code below.

```c#
// Add new() to TView constraints
public static IServiceCollection AddTransient<TView, TViewModel>(this IServiceCollection services)
    where TView : BindableObject, new()
    where TViewModel : class, INotifyPropertyChanged
{
    /* Current Implementation
    return services
        .AddTransient<TViewModel>()
        .AddTransient<TView>();
    */

    // Modified implementation
    return services
        .AddTransient<TViewModel>()
        .AddTransient(serviceProvider => new TView() { BindingContext = serviceProvider.GetService(typeof(TViewModel)) });
}
```

Adding that functionality would eliminate the need to inject a ViewModel instance in the View's code-behind.

**Comment by CliffAgius** [comment]
Closed as per the Community Stand-up discussion here: https://youtu.be/N9wMcBP4jtg?t=2889
